src/Microsoft.ML.Data/Transforms/ValueMapping.cs

```diff
@@ -46,7 +46,22 @@ public class ValueMappingEstimator : TrivialEstimator<ValueMappingTransformer>
         /// <param name="columns">The list of names of the input columns to apply the transformation, and the name of the resulting column.</param>
         internal ValueMappingEstimator(IHostEnvironment env, IDataView lookupMap, string keyColumn, string valueColumn, params (string outputColumnName, string inputColumnName)[] columns)
             : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(ValueMappingEstimator)),
-                    new ValueMappingTransformer(env, lookupMap, keyColumn, valueColumn, columns))
+                  new ValueMappingTransformer(env, lookupMap, lookupMap.Schema[keyColumn], lookupMap.Schema[valueColumn], columns))
+        {
+            _columns = columns;
+        }
+
+        /// <summary>
+        /// Constructs the ValueMappingEstimator, key type -> value type mapping
+        /// </summary>
+        /// <param name="env">The environment to use.</param>
+        /// <param name="lookupMap">An instance of <see cref="IDataView"/> that contains the key and value columns.</param>
+        /// <param name="keyColumn">Name of the key column in <paramref name="lookupMap"/>.</param>
+        /// <param name="valueColumn">Name of the value column in <paramref name="lookupMap"/>.</param>
+        /// <param name="columns">The list of names of the input columns to apply the transformation, and the name of the resulting column.</param>
+        internal ValueMappingEstimator(IHostEnvironment env, IDataView lookupMap, DataViewSchema.Column keyColumn, DataViewSchema.Column valueColumn, params (string outputColumnName, string inputColumnName)[] columns)
```

Comment:
If we are adding this constructor, is the one above (that takes a key column name and value column name) needed?

Reply:
+1, especially since the extension that calls into the ctor, takes in DataViewSchema.Column and passes down only the name of the column. 

---
In reply to: [271533565](https://github.com/dotnet/machinelearning/pull/3180#discussion_r271533565) [](ancestors = 271533565)

Reply:
I will remove it in the next iteration.

---
In reply to: [271577343](https://github.com/dotnet/machinelearning/pull/3180#discussion_r271577343) [](ancestors = 271577343,271533565)